subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java

```diff
@@ -812,6 +836,23 @@ public void rebuildTransitions(final List<SubscriptionBaseEvent> inputEvents, fi
 
         removeEverythingPastCancelEvent(events);
 
+        transitions = new LinkedList<SubscriptionBaseTransition>();
+        transitionsWithDeletedEvents = new LinkedList<SubscriptionBaseTransition>();
+
+        final List<SubscriptionBaseEvent> activeEvents = inputEvents.stream().filter(event -> event.isActive()).collect(Collectors.toList());
+        rebuildTransitionsInternal(activeEvents, catalog, true);
```

Comment:
I think calling `rebuildTransitionsInternal` twice because we have to build 2 list, i.e. `transitions` and `transitionsWithDeletedEvents` makes sense, but I am not convinced
doing a first pass with just the active events and then if `includeDeletedEvents` is true doing a second pass to complete the list of `transitionsWithDeletedEvents` leads to the correct result as 'transitions' span events and they need to be ordered. Also, there is a lot of state between what `inputEvents` contains based on `includeDeletedEvents=true|false`, the state member variable `includeDeletedEvents` and the new boolean `activeEvents`.


Instead, could we:
1. If `includeDeletedEvents=false` (most common case), just call once `rebuildTransitionsInternal`, no filtering even required
2. If `includeDeletedEvents=true`, call `rebuildTransitionsInternal` once with filtering to build 'transitions'  and then call `rebuildTransitionsInternal` without filtering to build `transitionsWithDeletedEvents`.

Perhaps, we could make `rebuildTransitionsInternal` a static method and pass the correct List<SubscriptionBaseTransition> as an argument, i.e either `transitions` or `transitionsWithDeletedEvents` based on cases; it looks like you would need to pass `id`, `bundleId`, `bundleExternalKey` in order to do that. The benefit is that this method
would just build a list of transitons based on input and there is no additional boolean `activeEvents` nor use of  the state member variable `includeDeletedEvents` in this method anymore making it much simpler.